### ISSUE TITLE
Bulk Add from browser (only selected cards)

### ISSUE THREAD

**Issue by Nayelianne** [IllDepence/anki_add_pitch_plugin#35]
I don't know how tricky it would be to implement, but I would love a way to add pitch to specific cards while I'm creating them, be it by bulk from the edit menu on the browser, or by pressing a button on the card itself or through a right click menu when inside a card.

I know we have a button for manually adding pitches, but that means I have to go out and research the pitch accent for the card (which breaks the flow and can be time consuming), instead of having the addon automatically generate it.
Having such a feature would enable me to add pitches on a "as needed" basis, instead of having to mess with config files if I need it on only certain cards.

**Comment by IllDepence** [comment]
Hi, thanks for the suggestion.
I liked the idea and gave it a shot.

A few notes

* Compared to the bulk add, where you select all the relevant fields once and are done, I though it would be too cumbersome to do it again and again for single cards, so ideally it would be possible to either
    * (a) select the fields once and have the add-on remember the choice (would have to be deck and card type specific), or
    * (b) have the add-on figure out by itself what the expression and reading fields are
* In the end I went with the latter option, risking some lookup failures (although then there’s still manual mode), but giving users a bit more freedom (e.g. the possibility to add the pitch accent to different fields on a card by card basis—not sure if any one would want to do that though)
* Assumptions made in the current implementation:
    * The Japanese expression field comes before any other fields containing Japanese (e.g. example sentences)
    * Reading is always denoted in hiragana only (although it’s not required for a reading to be identified for the function to work)

I’m currently in the process of a bit of an overhaul of the add-on code and can’t say when I’ll update it on AnkiWeb, but you could install a version with the added functionality via *Tools→Addons→Install from file* and tell me

* if it works on your end
* if you noticed any problems/have additional suggestions

Here’s a [download link](https://moc.sirtetris.com/japanese_pitch_accent.ankiaddon) for the version with the added functionality (tested on Anki 2.1.49).

**Comment by Nayelianne** [comment]
Wow, that was so quick, I really appreciate all the work!
I played around with it for a bit and it's perfect for my needs, my Expression field is normally the first one on a card so that's totally fine by me. Not that I would mind if I had to mess with the config files to determine the field/note type, since many addons I use already have me doing that haha.

Love that you even made a little visual button for it! Very happy that I can more easily add pitches to different fields without having to open google.

I'm also on Anki 2.1.49 so sadly can't comment for other versions, but no bugs encountered here so far.
Thanks again <3

**Comment by Nayelianne** [comment]
**TL:DR; Found an addon incompatibility: Collapsible Fields.**
--
Hi again! I come with not so great news haha. Things were working as a wonder for a while.
Buuut eventually for some reason both the new automatic generation button and the manual buttons stopped working. When I clicked the button there was a one second pause like it was _trying_ to do something, but then nothing happened at all, no error message, no nothing. Bulk add still worked as intended despite that though. 

Sooo I did some digging around to what the cause may have been, and finally narrowed it down to an incompatible addon called ["Collapsible Fields" ](https://ankiweb.net/shared/info/1896168623)- that I started to use due to having an unholy amount of fields (121!) that a dictionary scraper addon ended up making me use. No fields were actually collapsed when the bug happened, so I guess it just interferes with yours by existing.

"Collapsible Fields" doesn't seem to interfere with any of my dictionary scraper addons that fill predermined fields by right click activation, nor the furigana add-on that also uses little buttons if that information is of any help at all.

I don't know if it's something worth looking into since it involves a different addon. But either way I figured you might like to know even if just to add to the compatibility documentation or in case someone comes pestering you about things not working, haha.
